### PR TITLE
Feature/508 Cypress Lighthouse integration

### PR DESCRIPTION
**Pull request by Elaine-Krauss-TCG**
#652

### CODE REVIEW

**Comment by Elaine-Krauss-TCG** [comment]
IMO, the 2 remaining code smells should stay as they keep typescript happy

File: front-end/cypress/support/commands.ts [hunk 1]
```diff
@@ -204,6 +206,31 @@ export function calendarSetValue(calendar: string, dateObj: Date = new Date()) {
   cy.shortWait();
 }
 
+export function runLighthouse(directory: string, filename: string) {
+  cy.lighthouse(
+    {
+      performance: 0,
+      accessibility: 90,
```
Comment (dheitzer) [inline front-end/cypress/support/commands.ts:213]: Might be worth a discussion what we want to set this number to? Maybe we could also discuss if we want to set/track any of the other thresholds.

File: front-end/cypress/e2e/general/lighthouse.cy.ts [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+  cy.deleteAllReports();
+  cy.medWait();
+  cy.visit('/reports');
+  const report = generateReportObject();
+  cy.createReport(report);
+  cy.navigateToTransactionManagement();
+  cy.shortWait();
+}
+
+describe('Generates lighthouse reports for every page', () => {
```
Comment (dheitzer) [inline front-end/cypress/e2e/general/lighthouse.cy.ts:18]: This is great

File: front-end/cypress/support/commands.ts [hunk 3]
```diff
@@ -204,6 +206,31 @@ export function calendarSetValue(calendar: string, dateObj: Date = new Date()) {
...
+      screenEmulation: {
+        mobile: false,
+        disable: false,
+        width: Cypress.config('viewportWidth'),
+        height: Cypress.config('viewportHeight'),
+        deviceScaleRatio: 1,
+      },
+    }
+  ).then(() => {
+    cy.exec(`mv lighthouse.html cypress/lighthouse/${directory}/${filename}.html`);
```
Comment (dheitzer) [inline front-end/cypress/support/commands.ts:230]: I wonder if we should add a timestamp to the filename or directory

File: front-end/cypress/support/commands.ts (hunk 1 above)
Comment (mjtravers) [reply front-end/cypress/support/commands.ts:213]: We can discuss later on after it's been running for a bit and revisit the configuration settings.

File: front-end/cypress/support/commands.ts (hunk 3 above)
Comment (mjtravers) [reply front-end/cypress/support/commands.ts:230]: We'll be managing the report dates via the CircleCI process so, at least at this point, we can forgo the timestamps.

**Review by mjtravers — APPROVED** [review]

**Pull request merged**